synapse/lib/node.py

```diff
@@ -327,8 +327,19 @@ async def set(self, name, valu, init=False):
             if curv == norm:
                 return False
 
-        props = {prop.name: norm}
-        nodeedits = await self.snap.getNodeAdds(self.form, self.ndef[1], props, addnode=False)
+        # Optimize fast property setting when the destination property is
+        # not a form and when normalizing the property we have no subs
+        # or adds to consider.
+        if self.snap.core.model.form(prop.type.name) is None \
+                and info.get('subs') is None \
+                and not info.get('adds') \
+                and not prop.locked:
```

Comment:
We should probably move this check specifically to somewhere near the entry of the function since we'd want to bail on it either way